voice/src/main/kotlin/VoiceConnectionBuilder.kt

```diff
@@ -166,19 +167,23 @@ public class VoiceConnectionBuilder(
             .build()
         val udpSocket = udpSocket ?: GlobalVoiceUdpSocket
         val audioProvider = audioProvider ?: EmptyAudioPlayerProvider
-        val nonceStrategy = nonceStrategy ?: LiteNonceStrategy()
+        val encryption = if ((receiveVoice || streams != null) && encryption?.supportsDecryption != true) {
+            VoiceEncryption.XSalsaPoly1305()
+        } else {
+            encryption ?: VoiceEncryption.AeadAes256Gcm
+        }
```

Comment:
It might be useful to log if the user's chosen encryption method ends up not being used.

Reply:
or even throw an exception? but logging a warning is probably best.